lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -602,26 +602,51 @@ def write(chunk)
 
           rescue GRPC::BadStatus => error
             increment_failed_requests_count(error.code)
+            # See the mapping between http status and grpc status code at:
+            # https://github.com/grpc/grpc/blob/master/src/core/lib/transport/status_conversion.cc
             case error.code
-            when GRPC::Core::StatusCodes::CANCELLED,
-                 GRPC::Core::StatusCodes::UNAVAILABLE,
-                 GRPC::Core::StatusCodes::DEADLINE_EXCEEDED,
-                 GRPC::Core::StatusCodes::INTERNAL,
-                 GRPC::Core::StatusCodes::UNKNOWN
-              # Server error, so retry via re-raising the error.
+            # Server error, so retry via re-raising the error.
+            when \
+                # Maps to http status 500 (Internal Server Error)
```

Comment:
[Optional] The "Maps to" part is repetitive and seems redundant.

Reply:
Removed.